model/src/main/kotlin/config/OrtConfiguration.kt

```diff
@@ -178,8 +182,12 @@ data class OrtConfiguration(
 
             val configResult = loader.loadConfig<OrtConfiguration>(prefix = "ort")
             val config = configResult.getOrElse { failure ->
-                require(sources.isEmpty()) {
-                    "Failed to load ORT configuration: ${failure.description()}"
+                val isFailureDueToEmptyConfig = failure is ConfigFailure.UndefinedTree
+                    || (failure is ConfigFailure.MissingConfigValue && sources.isEmpty())
```

Comment:
commit message nit: As I read the, the change intends to only reduce the cases of silently ignoring things.
This makes me wonder why `failure is ConfigFailure.UndefinedTree || ` is added, which looks as if there could be cases now silently ignored, which previously weren't. Should this bit be clarified?

apart from that: Should this change be marked breaking? with `!`

Reply:
> Should this bit be clarified?

I've tried to improve the commit message in that regard.

> Should this change be marked breaking? with `!`

I don't think so. This change is not breaking correct user-facing behavior from before, but uncovering cases where previously the default configuration was erroneously used.